src/common/AppToggle.vue

```diff
@@ -0,0 +1,102 @@
+<template>
+  <label class="container">
+    <input
+      v-bind="$attrs"
+      class="input"
+      type="checkbox"
+      :checked="modelValue"
+      @change="$emit('update:modelValue', $event.target.checked)"
+    />
+
+    <span class="switch">
+      <span :class="['trigger', { disabled: modelValue }]">{{ leftText }}</span>
+      <span :class="['trigger', { disabled: !modelValue }]">{{
+        rightText
+      }}</span>
+    </span>
+  </label>
+</template>
+
+<script>
+export default {
+  props: {
+    leftText: {
+      type: String,
+      required: true
+    },
+    rightText: {
+      type: String,
+      required: true
+    },
+    modelValue: {
+      type: Boolean,
+      required: true
+    }
+  },
+  emits: ["update:modelValue"]
+};
+</script>
+
+<style scoped>
+.container {
+  cursor: pointer;
+  display: flex;
+  box-sizing: border-box;
+  min-width: 20rem;
+}
+
+.trigger {
+  flex: 1;
```

Comment:
The key here is this `flex: 1` property.

When you assign multiple flex elements the same `flex` number, those elements will always take up the same proportion of space. Think of it like fractions — with two elements having `flex: 1`, each element will take up `1 (the flex number) / 2 (total elements)` space. When you have three elements, each element will take up `1/3` space and so on.

In this case, the elements will equally take up 50% of the container's space.

For more: https://developer.mozilla.org/en-US/docs/Web/CSS/flex

Reply:
thank you! I will definitely take a look at the document